lib/primitives/keyring.js

```diff
@@ -154,19 +155,25 @@ class KeyRing {
    * Inject data from public key.
    * @private
    * @param {Buffer} key
+   * @param {Boolean?} compress
+   * @returns {KeyRing}
    */
 
-  fromPublic(key) {
+  fromPublic(key, compress) {
     assert(Buffer.isBuffer(key), 'Public key must be a buffer.');
     assert(secp256k1.publicKeyVerify(key), 'Not a valid public key.');
-    this.publicKey = key;
```

Comment:
Style nit: since the `if`/`else` blocks are only 1 line each, its possible to omit the curly braces to make it easier to read overall
```javascript
    if (compress === undefined)
      this.publicKey = key;
    else
      this.publicKey = secp256k1.publicKeyConvert(key, compress !== false);
```

Also can you simplify the `compress !== false` part? We know that compress is not `undefined`
and we expect it to be a `boolean` so I think that something like

```javascript
if (compress)
  assert(typeof compress === 'boolean');
```

should allow you to do something like

```javascript
this.publicKey = secp256k1.publicKeyConvert(key, compress);
```

which makes the code way more simple to read

Reply:
Do you think this is easy to read?

```javascript
if (typeof compress === 'boolean')
  this.publicKey = secp256k1.publicKeyConvert(key, compress);
else
  this.publicKey = key;
```

Reply:
Yes, it's indeed better.